hail/src/main/scala/is/hail/expr/types/physical/ComplexPType.scala

```diff
@@ -3,7 +3,7 @@ package is.hail.expr.types.physical
 import is.hail.annotations.UnsafeOrdering
 
 abstract class ComplexPType extends PType {
-  val representation: PType
+  def representation: PType
```

Comment:
this can have bad consequences. Would prefer to leave as val.

Reply:
Done. The consequence you're speaking about, you mean to indicate to the inheritor that this should be immutable rather than potentially dynamic?

Reply:
Oh! The reason I did this is because otherwise there is not, as far as I could find, a neat way of indicating that the PInterval representation is a PStruct

The way I handled it is by defining an unimplemented PStruct

```scala
def representation: PStruct = ???
```

and then overriding it in PCanonicalInterval

```scala
override val representation: PStruct = PStruct(
      required,
      "start" -> pointType,
      "end" -> pointType,
      "includesStart" -> PBooleanRequired,
      "includesEnd" -> PBooleanRequired)
```

Reply:
I tried a number of ways, including changing the type of representation to a type that PInterval overrode, but this wouldn't type check.

The other option is to cast everywhere